code-challenges.test.js

```diff
@@ -14,39 +14,214 @@
+
+--- AFTER TESTING
+4. After running some tests, I realized I need to add another step to this to account for case sensitivity with the input string. So, I'll use an OR operator and .toLowerCase() on both the input array element and the input string to account for this.
+*/
+
+const findLetterInElement = (arr, str) => {
+  return arr.filter((v) => v.toLowerCase().includes(str.toLowerCase()));
+};
+
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("checkFullHouse", () => {
+  it(`Takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.`, () => {
+    const hand1 = [5, 5, 5, 3, 3];
+    // Expected output: true
+    const hand2 = [5, 5, 3, 3, 4];
+    // Expected output: false
+    const hand3 = [5, 5, 5, 5, 4];
+    // Expected output: false
+    const hand4 = [7, 2, 7, 2, 7];
+    // Expected output: true
+    const hand5 = [5, 5, 5, 3, 3, 6];
+    // Expected output: "A full house cannot be more than 5 cards."
+    const hand6 = [5, 5, 5, 3];
+    // Expected output: "A full house cannot be more than 5 cards."
+    expect(checkFullHouse(hand1)).toEqual(true);
+    expect(checkFullHouse(hand2)).toEqual(false);
+    expect(checkFullHouse(hand3)).toEqual(false);
+    expect(checkFullHouse(hand4)).toEqual(true);
+    expect(checkFullHouse(hand5)).toEqual(
+      "A full house cannot be more than 5 cards."
+    );
+    expect(checkFullHouse(hand6)).toEqual(
+      "A full house cannot be more than 5 cards."
+    );
+  });
+});
+
+// GOOD FAIL:  ReferenceError: checkFullHouse is not defined
 
 // b) Create the function that makes the test pass.
+
+/*
+--------- FUNCTION INFORMATION:
+Describe what the function will do: Function will take in an array of strings, then check against its values to determine if it's a full house.
+Function Name: checkFullHouse
+Function Parameter(s): 1, array of strings
+--------- INPUT / OUTPUT:
+input 1: array of strings
+output 1: boolean
+
+--------- RESEARCH PHASE: Before I start on this, I'm going to do some research on some different methods and steps. The array method .every() will return a boolean value based on a specified test, so I think that's what I'm going to use to iterate. A full house should never be larger than 5, so we can use that in a conditional. We want to determine if we have TWO matching instances of an element, one 3 and one 2.
+*/
+
+const checkFullHouse = (arr) => {
```

Comment:
Really cool approach and this approach will be helpful to keep in mind for more advanced problems as well in the future.  Great work 👏